Automattic/woocommerce-payments
# Sync blog locale to WCPay Account

Fixes 888-gh-Automattic/woocommerce-payments-server

#### Changes proposed in this Pull Request

Adds synchronization of the blog's site language with the WCPay account.

The reason for this PR is that the site language does not get synced when using Jetpack connection (sites that use WCPay only), but it does get synced when using the Jetpack plugin.

__NOTE:__ this PR needs 904-gh-Automattic/woocommerce-payments-server to be deployed in the server.

#### Testing instructions

Test that the language is stored in the server when setting up WooCommerce Payments (onboarding):

0. Checkout the corresponding `woocommerce-payments-server` branch: `add/887-add-locale-to-account`
1. Go to WCPay Dev, and make sure `Enable API request redirection` is checked
2. Go to Settings, and change the `Site Language` to other than `English (United States)` (needed because `en_US` is the server default)
3. Go to Payments, and complete the onboarding
4. Check in the server DB that the account locale was stored correctly

Test that the language is updated in the server when changing the site language:

0. Checkout the corresponding `woocommerce-payments-server` branch: `add/887-add-locale-to-account`
1. Go to WCPay Dev, and make sure `Enable API request redirection` is checked
2. Go to Settings, and change the `Site Language`
3. Check in the server DB that the account locale was changed

-------------------
